src/ess/reduce/nexus/workflow.py

```diff
@@ -204,102 +147,62 @@ def load_nexus_sample(
-    Load a NeXus source group from a file.
+def nx_class_for_monitor() -> NeXusClass[MonitorType]:
+    return NeXusClass[MonitorType](snx.NXmonitor)
 
-    Parameters
-    ----------
-    location:
-        Location spec for the source group.
-    """
-    return NeXusSource[RunType](nexus.load_component(location, nx_class=snx.NXsource))
 
+def nx_class_for_detector() -> NeXusClass[snx.NXdetector]:
+    return NeXusClass[snx.NXdetector](snx.NXdetector)
 
-def load_nexus_detector(
-    location: NeXusComponentLocationSpec[snx.NXdetector, RunType],
-) -> NeXusDetector[RunType]:
-    """
-    Load detector from NeXus, but with event data replaced by placeholders.
 
-    As the event data can be large and is not needed at this stage, it is replaced by
-    a placeholder. A placeholder is used to allow for returning a scipp.DataArray, which
-    is what most downstream code will expect.
-    Currently the placeholder is the detector number, but this may change in the future.
+def nx_class_for_source() -> NeXusClass[snx.NXsource]:
+    return NeXusClass[snx.NXsource](snx.NXsource)
 
-    The returned object is a scipp.DataGroup, as it may contain additional information
-    about the detector that cannot be represented as a single scipp.DataArray. Most
-    downstream code will only be interested in the contained scipp.DataArray so this
-    needs to be extracted. However, other processing steps may require the additional
-    information, so it is kept in the DataGroup.
 
-    Loading thus proceeds in three steps:
+def nx_class_for_sample() -> NeXusClass[snx.NXsample]:
+    return NeXusClass[snx.NXsample](snx.NXsample)
 
-    1. This function loads the detector, but replaces the event data with placeholders.
-    2. :py:func:`get_calibrated_detector` drops the additional information, returning
-       only the contained scipp.DataArray, reshaped to the logical detector shape.
-       This will generally contain coordinates as well as pixel masks.
-    3. :py:func:`assemble_detector_data` replaces placeholder data values with the
-       event data, and adds source and sample positions.
 
-    Parameters
-    ----------
-    location:
-        Location spec for the detector group.
+def load_nexus_component(
+    location: NeXusComponentLocationSpec[Component, RunType],
+    nx_class: NeXusClass[Component],
+) -> NeXusComponent[Component, RunType]:
     """
-    # The selection is only used for selecting a range of event data.
-    location = replace(location, selection=())
+    Load a NeXus component group from a file.
 
-    return NeXusDetector[RunType](
-        nexus.load_component(location, nx_class=snx.NXdetector, definitions=definitions)
-    )
-
-
-def load_nexus_monitor(
-    location: NeXusMonitorLocationSpec[RunType, MonitorType],
-) -> NeXusMonitor[RunType, MonitorType]:
-    """
-    Load monitor from NeXus, but with event data replaced by placeholders.
+    When loading a detector or monitor, event data is replaced by placeholders.
```

Comment:
Maybe I missed something, but is this for loading only detectors and monitors? I thought `Component` was also for sample and source?

**Edit**: I now see that the case for `NXsample` is overloaded by the `load_nexus_sample` function above. Are we missing a `load_nexus_source` function?

Reply:
It is also used for the source.  But this paragraph basically just applies to the groups that have event data.